python/triton/runtime/jit.py

```diff
@@ -478,7 +479,7 @@ def __init__(self, fn, version=None, do_not_specialize=None, debug=None, noinlin
 
         # function source code (without decorators)
         self.src = textwrap.dedent(inspect.getsource(fn))
-        self.src = self.src[self.src.find("def"):]
+        self.src = self.src[re.search(r"^def\s+\w+\s*\(", self.src, re.MULTILINE).start():]
```

Comment:
Oh I guess the `\w` is okay because we only match the name up until the open paren.

But why is `^` correct?

Reply:
Yeah I'm not matching against arg list anymore, because we probably just need to find the position of `def`. \w+ part correspond to the function name before the first ( in the parsing. `def` should always occur at the beginning of the line, otherwise it will be a indent error in Python. Also `def` is a keyword so no other syntax rule can use it.

Reply:
I was tempting to make it just re.search("^def ", ...) which I couldn't come up with a counter example, but to make a bit safer I add a bunch of further matching again the `foo_name(` part.

Reply:
Oh one thing I forgot to bring up: we always call textwrap.dedent() before this line, and since we always start a `def` with a new line, I feel `^` works. Happy to be corrected.